app/Console/Commands/RussianRoulette.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Console\Commands;
+
+
+use Illuminate\Console\Command;
+use Illuminate\Contracts\Cache\Repository as CacheRepository;
+
+
+class RussianRoulette extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'ru-roulette:play';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Play Russian Roulette game.';
+
+    private CacheRepository $cacheRepository;
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct(CacheRepository $cacheRepository)
+    {
+        parent::__construct();
+        $this->cacheRepository = $cacheRepository;
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+
+        $number = 0;
+        $winstat = [];
+        $losestat = [];
+        $counter = 0;
+        $numberstatplayer = [];
+        $numberstatgame = [];
+        $this->info("Welcome to 'THE RUSSIAN ROULETTE'!");
+        while ($number > 1 or $number < 6) {
+
+            $this->info("To continue please enter your number from 1 to 6:");
+            $number = $this->ask("Command: ");
+
+            if ($number < 1 or $number > 6) {
+                $this->info("Game over!");
+
+                break;
+            } else {
+                $numberstatplayer[] = $number;
+
+                $this->info("Your number:{$number}");
+                $game = random_int(1, 6);
+                array_push($numberstatgame, $game);
+                $this->info("Destiny has chosen: {$game} ");
+                $counter++;
+                if ($number == $game) {
+                    $this->info("Sorry you are dead");
+                    array_splice($losestat, ($counter - 1), 0, $number);
+                    array_splice($winstat, ($counter - 1), 0, 'out of count');
```

Comment:
What sense in array_splices here?

Reply:
It is for index addition in array. For example in $losestat will be added $number if lost and for $winstat we have to move from index 0 to 1 however since it is not a "win" we declare it as "out of count". Later in statistics we simply use array_count_values for both $losestats and $winstats and receive how many time there was a win/lose number